### PR TITLE
Add support for pre-existing OPSM

### PR DESCRIPTION
**Pull request by mslipper**
This PR adds support for deploying OP Chains against an existing OPSM deployment. It adds a new `OPSMAddress` field to the intent to hold the address of the OPSM. When specified, the `ContractsRelease` field is ignored. In the future, this field will be pulled from the Superchain Registry.

Since the Go Forge tooling doesn't support forking yet, the deployment is performed using a raw call to `eth_sendRawTransaction`. Data about the Superchain deployments is pulled from the OPSM itself via `eth_call`. To expose the deployment output following the Superchain deployment, I updated the `Deployed` event to emit the ABI-encoded bytes of the `DeployOutput` struct to avoid stack-too-deep errors. This isn't ideal, but at least it gets me all of the fields I need in a log event without more invasive changes.

Closes https://github.com/ethereum-optimism/platforms-team/issues/285.

### CODE REVIEW

**Review by mds1 — APPROVED** [review]

File: op-chain-ops/deployer/broadcaster/broadcaster.go [hunk 1]
```diff
@@ -11,6 +11,7 @@ import (
 type Broadcaster interface {
 	Broadcast(ctx context.Context) ([]BroadcastResult, error)
 	Hook(bcast script.Broadcast)
+	Sender() common.Address
```
Comment (protolambda) [inline op-chain-ops/deployer/broadcaster/broadcaster.go]: I don't like this change as much. Broadcast structs can have different senders, the `vm.broadcast` family of cheatcodes allows you to mock different sender addresses throughout a script. There shouldn't be a single enshrined sender.

File: op-chain-ops/deployer/opcm/contract.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+type Contract struct {
+	addr   common.Address
+	client *ethclient.Client
+}
```
Comment (protolambda) [inline op-chain-ops/deployer/opcm/contract.go:17]: This is fine for now, but lets' separate remote-client and local deployer code, once we have forking functionality.

File: op-chain-ops/deployer/broadcaster/broadcaster.go (hunk 1 above)
Comment (mslipper) [reply op-chain-ops/deployer/broadcaster/broadcaster.go]: I can rename to `Broadcaster()` if that helps. This is used to represent who the underlying private key is.

File: op-chain-ops/deployer/pipeline/init.go [hunk 3]
```diff
@@ -34,6 +35,37 @@ func Init(ctx context.Context, env *Env, artifactsFS foundry.StatDirFs, intent *
...
+		env.Logger.Debug(
+			"populating protocol versions and superchain config addresses",
+			"protocolVersions", protocolVersions.Hex(),
+			"superchainConfig", superchainConfig.Hex(),
+		)
+
+		st.SuperchainDeployment = &state.SuperchainDeployment{
+			ProtocolVersionsProxyAddress: protocolVersions,
+			SuperchainConfigProxyAddress: superchainConfig,
+		}
```
Comment (protolambda) [inline op-chain-ops/deployer/pipeline/init.go:66]: Comments here about the unused struct fields, and why they are left empty, would be nice here.

File: op-chain-ops/deployer/broadcaster/broadcaster.go (hunk 1 above)
Comment (protolambda) [reply op-chain-ops/deployer/broadcaster/broadcaster.go]: But `script.Broadcast` already has a `From` field. Wouldn't that conflict, if the address does not match? Maybe a "default broadcaster" makes sense to have, but generally the script decides who broadcasts, and the one who set up the hook should error if it doesn't have the key to do something with it.

Comment (mslipper) [reply op-chain-ops/deployer/broadcaster/broadcaster.go]: In the case of the keyed broadcaster the `script.Broadcast`'s `From` field can already be different from the private key in the broadcaster itself. Agree we should error if they don't match - mind if I log this as a ticket?

Comment (mslipper) [reply op-chain-ops/deployer/broadcaster/broadcaster.go]: I'll remove the interface method for now to avoid confusion.

Comment (protolambda) [reply op-chain-ops/deployer/broadcaster/broadcaster.go]: A ticket for key mismatch with broadcast request is fine. Thanks

**Review by protolambda — APPROVED** [review]

**Comment by protolambda** [comment]
Opened a ticket to track the broadcast sender <> deployer key check: https://github.com/ethereum-optimism/optimism/issues/12127

**Pull request merged**